pkg/extensions/customresources.go

```diff
@@ -106,7 +107,14 @@ func WaitUntilObjectReadyWithHealthFunction(
 
 		if err := healthFunc(obj); err != nil {
 			lastObservedError = err
-			logger.WithError(err).Errorf("%s did not get ready yet", extensionKey(kind, namespace, name))
+
+			switch logger := log.(type) {
+			case logrus.FieldLogger:
+				logger.WithError(err).Errorf("%s did not get ready yet", extensionKey(kind, namespace, name))
+			case logr.Logger:
+				logger.Error(err, "Object did not get ready yet", "object", extensionKey(kind, namespace, name))
```

Comment:
The passed logger should already have the object's name and namespace attached and the kind is implied by the controller name.
Hence, we don't need to add them here again
```suggestion
				logger.Error(err, "Object did not get ready yet")
```